File: ricemill/utils/manufacturing/operation/operation.py
import frappe
from frappe.custom.doctype.custom_field.custom_field import create_custom_fields
def operation_customize_field():
    custom_fields = {
        "Operation": [
            dict(fieldname='section_break_opration', label='Operation',
                fieldtype='Section Break',insert_after='quality_inspection_template',collapsible=1),
            dict(fieldname="grading",
            label="Grading",
            fieldtype="Check",
            insert_after="section_break_opration",
            ),

            dict(fieldname="capacity_operation",
            label="Max Capacity",
            fieldtype="Data",
            insert_after="grading",
            depends_on="eval:doc.grading==1",
            mandatory_depends_on="eval:doc.grading==1"
            ),
            dict(fieldname="capacity_exceeds",
            label="When Capacity exceeds",
            fieldtype="Select",
            options="\nAllow\nBlock",
            insert_after="capacity_operation",
            depends_on="eval:doc.grading_operation==1",
            mandatory_depends_on="eval:doc.grading_operation==1"
            ),
            dict(fieldname="soaking",
            label="Soaking",
            fieldtype="Check",
            insert_after="capacity_exceeds",
            ),
            dict(fieldname="boiling",
            label="Boiling",
            fieldtype="Check",
            insert_after="soaking",
            ),
            dict(fieldname="dryer",
            label="Dryer",
            fieldtype="Check",
            insert_after="boiling",
            ),
            dict(fieldname="hulling",
            label="Hulling",
            fieldtype="Check",
            insert_after="dryer",
            )
            
            ]
            }
    create_custom_fields(custom_fields)

File: ricemill/ricemill/custom/js/operation.js
frappe.ui.form.on("Operation",{
    refresh:function(frm){
        cur_frm.fields_dict.grading.$input.on("click", function() {
            frm.set_value('soaking',0)
            frm.set_value('boiling',0)
            frm.set_value('dryer',0)
            frm.set_value('hulling',0)
        }),
        cur_frm.fields_dict.soaking.$input.on("click", function() {
            frm.set_value('grading',0)
            frm.set_value('boiling',0)
            frm.set_value('dryer',0)
            frm.set_value('hulling',0)
    }),
        cur_frm.fields_dict.boiling.$input.on("click", function() {
            frm.set_value('grading',0)
            frm.set_value('soaking',0)
            frm.set_value('dryer',0)
            frm.set_value('hulling',0)
    }),
        cur_frm.fields_dict.dryer.$input.on("click", function() {
            frm.set_value('grading',0)
            frm.set_value('soaking',0)
            frm.set_value('boiling',0)
            frm.set_value('hulling',0)
    }),        
        cur_frm.fields_dict.hulling.$input.on("click", function() {
            frm.set_value('grading',0)
            frm.set_value('soaking',0)
            frm.set_value('boiling',0)
            frm.set_value('dryer',0)
        })
    }
})
   

